homeassistant/components/sensor/homematicip.py

```diff
@@ -0,0 +1,418 @@
+    if home.devices is None:
+        return False
+    for device in home.devices:
+        add_devices([HomematicipDeviceStatus(hass, home, device)])
+        if isinstance(device, HeatingThermostat):
+            add_devices([HomematicipHeatingThermostat(hass, home, device)])
+        if isinstance(device, TemperatureHumiditySensorWithoutDisplay):
+            add_devices([HomematicipSensorThermometer(hass, home, device)])
+            add_devices([HomematicipSensorHumidity(hass, home, device)])
+        if isinstance(device, TemperatureHumiditySensorDisplay):
+            add_devices([HomematicipSensorThermometer(hass, home, device)])
+            add_devices([HomematicipSensorHumidity(hass, home, device)])
+    return True
+
+
+class HomematicipGenericDevice(Entity):
+    """Representation of an HomematicIP generic device."""
+
+    def __init__(self, hass, home, device, signal=None):
+        """Initialize the generic device."""
+        self.hass = hass
+        self._home = home
+        self._device = device
+
+        @callback
+        def event_device(event):
+            """Handle device state changes."""
+            _LOGGER.debug('Event Device: %s', self._device.label)
+            self.schedule_update_ha_state(True)
+
+        hass.bus.listen(EVENT_DEVICE_CHANGED, event_device)
+
+    def _name(self, addon=''):
+        """Return the name of the device."""
+        name = ''
+        if self._home.label != '':
+            name += self._home.label + ' '
+        name += self._device.label
+        if addon != '':
+            name += ' ' + addon
+        return name
+
+    @property
+    def name(self):
+        """Return the name of the generic device."""
+        return self._name()
+
+    @property
+    def icon(self):
+        """Return the icon of the generic device."""
+        return 'mdi:hexagon-outline'
+
+    @property
+    def state(self):
+        """Return the state of the generic device."""
+        return 'empty'
+
+    @property
+    def force_update(self):
+        """Force update."""
+        return True
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return True
+
+    @property
+    def available(self):
+        """Device available."""
+        return True
+
+    def _generic_state_attributes(self):
+        """Return the state attributes of the generic device."""
+        attr = {}
+        if hasattr(self._home.label, 'label'):
+            attr.update({ATTR_HOME: self._home.label})
+        if hasattr(self._device, 'label'):
+            attr.update({ATTR_LABEL: self._device.label})
+        if hasattr(self._device, 'lastStatusUpdate'):
+            last = self._device.lastStatusUpdate
+            if last is not None:
+                attr.update({ATTR_LASTSTATUS_UPDATE:
+                             last.isoformat()})
+        if hasattr(self._device, 'homeId'):
+            attr.update({ATTR_HOME_ID: self._device.homeId})
```

Comment:
This is an _extremely_ inefficient way of updating a dictionary if you're just setting 1 piece of info.

```python
attr[ATTR_HOME_ID] = self._device.homeId
```

I also don't think that homeId should be part of each attribute. Remove it.